### PR TITLE
chore: clean up in-code uk translations | FC-55

### PR DESCRIPTION
**Pull request by OmarIthawi**
Clean up after #422 

`make pull_translations` should be used

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @OmarIthawi! Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Comment by volodymyr-chekyrta** [comment]
@OmarIthawi, could you please fix the tests? You've deleted files, but something is still looking for them.
```
The following build commands failed:
	CopyStringsFile /Users/runner/Library/Developer/Xcode/DerivedData/OpenEdX-bsdxmseilanfjjcjomwdntdeffpi/Build/Products/DebugDev-iphonesimulator/Core.framework/uk.lproj/Localizable.strings /Users/runner/work/openedx-app-ios/openedx-app-ios/Core/Core/uk.lproj/Localizable.strings (in target 'Core' from project 'Core')
```

**Comment by OmarIthawi** [comment]
> @OmarIthawi, could you please fix the tests? You've deleted files, but something is still looking for them.
> 
> ```
> The following build commands failed:
> 	CopyStringsFile /Users/runner/Library/Developer/Xcode/DerivedData/OpenEdX-bsdxmseilanfjjcjomwdntdeffpi/Build/Products/DebugDev-iphonesimulator/Core.framework/uk.lproj/Localizable.strings /Users/runner/work/openedx-app-ios/openedx-app-ios/Core/Core/uk.lproj/Localizable.strings (in target 'Core' from project 'Core')
> ```

I tried to fix the failing test but I'm unsure how to find the step `CopyStringsFile`. Could you please help?

Please try to checkout the branch, run `Project -> Clean` and it should fix the issue. I don't have XCode on my Linux machine which makes this not possible.

**Comment by OmarIthawi** [comment]
> @OmarIthawi, could you please fix the tests? You've deleted files, but something is still looking for them.
> 
> ```
> The following build commands failed:
> 	CopyStringsFile /Users/runner/Library/Developer/Xcode/DerivedData/OpenEdX-bsdxmseilanfjjcjomwdntdeffpi/Build/Products/DebugDev-iphonesimulator/Core.framework/uk.lproj/Localizable.strings /Users/runner/work/openedx-app-ios/openedx-app-ios/Core/Core/uk.lproj/Localizable.strings (in target 'Core' from project 'Core')
> ```

I tried to fix the failing test but I'm unsure how to find the step `CopyStringsFile`. Could you please help?

@volodymyr-chekyrta Please try to checkout the branch, run `Project -> Clean` and it should fix the issue. I don't have XCode on my Linux machine which makes this not possible.

**Comment by volodymyr-chekyrta** [comment]
> > @OmarIthawi, could you please fix the tests? You've deleted files, but something is still looking for them.
> > ```
> > The following build commands failed:
> > 	CopyStringsFile /Users/runner/Library/Developer/Xcode/DerivedData/OpenEdX-bsdxmseilanfjjcjomwdntdeffpi/Build/Products/DebugDev-iphonesimulator/Core.framework/uk.lproj/Localizable.strings /Users/runner/work/openedx-app-ios/openedx-app-ios/Core/Core/uk.lproj/Localizable.strings (in target 'Core' from project 'Core')
> > ```
> 
> I tried to fix the failing test but I'm unsure how to find the step `CopyStringsFile`. Could you please help?
> 
> @volodymyr-chekyrta Please try to checkout the branch, run `Project -> Clean` and it should fix the issue. I don't have XCode on my Linux machine which makes this not possible.

I tried this, but the project build fails.
The files are not deleted correctly because the project files are still referenced; you have to remove them from XCode.
Does anyone on your team have XCode? Or maybe I can help you with this.
image

**Comment by OmarIthawi** [comment]
@volodymyr-chekyrta please help, that would be really appreciated. 

We only need to delete those files now they're backed to Transifex and I'm filling up gaps whenever I can.

Our iOS engineer is juggling a couple of tasks already and I need her focused on SAML SSO :)

**Comment by volodymyr-chekyrta** [comment]
> @volodymyr-chekyrta please help, that would be really appreciated.
> 
> We only need to delete those files now they're backed to Transifex
> 
> Our iOS engineer is juggling a couple of tasks already and I need her focused on SAML SSO :) and I'm filling up gaps whenever I can.

Could you please give me write permissions to this branch in your repo?
I'll fix it and push changes

**Comment by OmarIthawi** [comment]
@volodymyr-chekyrta you now have write access to the branches. I think we need to revert the change and remove the files manually again via the XCode UI.

**Comment by volodymyr-chekyrta** [comment]
@OmarIthawi Okay, that approach won't work :(
If we remove links from the translation files and then pull them via `make pull_translations`, files will be physically on the disk, but the project will not know about them.
image
image

**Comment by OmarIthawi** [comment]
>  Okay, that approach won't work :(

@volodymyr-chekyrta That's an issue. How did we manage the `tx pull` in previous versions of the application such as:
- https://github.com/openedx-unsupported/edx-app-ios/commits/master/Source/ar.lproj/Localizable.strings

I suppose people didn't just manually edit those files, did they?

**Comment by OmarIthawi** [comment]
@volodymyr-chekyrta I've checked about the issue more and I have some ideas:

### Option 1: Use xcString files
I'm not an iOS expert, but if the xcstrings can fit all translations in a single file, I suppose that'll help.

 - Apple demonstration video: https://developer.apple.com/videos/play/wwdc2023/10155/
 - Example Localizable.xcstrings file: https://github.com/glushchenko/fsnotes/blob/814948cdd46091739b7c975bf86f129ba32c8ca5/FSNotes/Localizable.xcstrings#L6-L29

Which is as far as I understand, uses one JSON file for all languages.

### Option 2: Automate adding the files
The other alternative it to automatically add the `uk` and other languages after pulling them via Transifex. This can be done via libraries like https://github.com/kronenthaler/mod-pbxproj

**Comment by OmarIthawi** [comment]
@volodymyr-chekyrta I've made some progress regarding the XCode project file edits, I'll pickup the work again in a day or two.

File: Authorization/Authorization.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -288,16 +287,6 @@
...
 
-/* Begin PBXHeadersBuildPhase section */
-		0770DE3628D0A319006D8A5D /* Headers */ = {
-			isa = PBXHeadersBuildPhase;
-			buildActionMask = 2147483647;
-			files = (
-			);
-			runOnlyForDeploymentPostprocessing = 0;
-		};
-/* End PBXHeadersBuildPhase section */
```
Comment (OmarIthawi) [inline Authorization/Authorization.xcodeproj/project.pbxproj:299]: @volodymyr-chekyrta is this concerning?

File: Authorization/Authorization.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -306,7 +295,6 @@
 				737436DC910C5A48E4C01AC7 /* [CP] Check Pods Manifest.lock */,
 				07169450296D913300E3DED6 /* Sources */,
 				07169451296D913300E3DED6 /* Frameworks */,
-				07169452296D913300E3DED6 /* Resources */,
```
Comment (OmarIthawi) [inline Authorization/Authorization.xcodeproj/project.pbxproj:309]: I'm unsure what's being removed here by the https://github.com/kronenthaler/mod-pbxproj package. It's automatic.

**Review by OmarIthawi — COMMENTED** [review]
@volodymyr-chekyrta this is ready for another review.

**Comment by volodymyr-chekyrta** [comment]
@OmarIthawi, seems the script adding files to the wrong location
Screenshot 2024-06-25 at 9 40 41

**Comment by volodymyr-chekyrta** [comment]
It should be like this
image

**Comment by OmarIthawi** [comment]
That's a bumper, a pretty bad one. Thanks for checking @volodymyr-chekyrta. 

I'll study the problem more and get back to you. I'm not ready to give up yet 😅

**Comment by OmarIthawi** [comment]
@brian-smith-tcril @volodymyr-chekyrta Hello again! I think this pull request is ready for merge.

Please take a look and let me know if you have any notes or questions.

**Comment by volodymyr-chekyrta** [comment]
> @brian-smith-tcril @volodymyr-chekyrta Hello again! I think this pull request is ready for merge.
> 
> Please take a look and let me know if you have any notes or questions.

Hi @OmarIthawi, sorry for the delay, I'll check it now

**Comment by volodymyr-chekyrta** [comment]
@OmarIthawi, very last thing, and we are good to go.
Could you please remove as well this file:
https://github.com/Zeit-Labs/openedx-app-ios-contrib/blob/remove-uk/OpenEdX/uk.lproj/Localizable.strings
image

And then add it during the `make pull_translations` script.

This file is empty, but this is the way to tell the system that we support that language
https://github.com/user-attachments/assets/f54610ad-cc81-47cc-a5f3-57aee4bea7d5

Thank you 🙏

**Comment by OmarIthawi** [comment]
Thanks @volodymyr-chekyrta, I'll do that.

**Comment by volodymyr-chekyrta** [comment]
@OmarIthawi, could you please tell me if PR is ready for the final pass?
I still see the uk file at OpenEdX/uk.lproj/Localizable.strings
image

**Comment by OmarIthawi** [comment]
Not ready yet @volodymyr-chekyrta. Still working on the last bit.

**Comment by OmarIthawi** [comment]
@volodymyr-chekyrta, it's currently ready for testing.

I've used git diffs and it seemed okay. Please let me know if it looks good in XCode.

I see that we're still missing plurals and countries.json localization. So, let's see what can we improve and whether XCStrings would be used eventually.

**Review by volodymyr-chekyrta — APPROVED** [review]
LGTM

**Comment by OmarIthawi** [comment]
Thanks @volodymyr-chekyrta for the review. 

I don't have merge access, I appreciate a merge to avoid future git conflics.

**Comment by openedx-webhooks** [comment]
@OmarIthawi 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/openedx-app-ios&entry.867055334=https://github.com/openedx/openedx-app-ios/pull/465&entry.1484655318=https://github.com/OmarIthawi&entry.752974735=2024-06-13+21:07&entry.1917517419=2024-08-19+07:23&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**